core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -911,7 +954,22 @@ object DynamicListenerConfig {
   )
 }
 
-class DynamicClientQuotaCallback(server: KafkaBroker) extends Reconfigurable {
+trait DynamicClientQuotaCallbackServer {
```

Comment:
We don't really need a trait for this, do we? It's just two objects that we can pass to the constructor of `DynamicClientQuotaCallback`